### PR TITLE
crimson: drop store from ECBackend to not shadow PGBackend::store

### PR DESCRIPTION
**Pull request by rzarzynski**
This problem was leading to crashes like this one:

```
../src/crimson/os/futurized_store.h:113:41: runtime error: member access within misaligned address 0xbebebebebebebebe for type 'struct Shard', which requires 8 byte alignment
0xbebebebebebebebe: note: pointer points here
<memory cannot be printed>
--Type <RET> for more, q to quit, c to continue without paging--

Thread 1 "crimson-osd" received signal SIGSEGV, Segmentation fault.
0x000055558e7a1dca in crimson::os::FuturizedStore::Shard::do_transaction (txn=..., ch=..., this=0xbebebebebebebebe) at ../src/crimson/os/futurized_store.h:113
113	      return do_transaction_no_callbacks(
(gdb) bt
```

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Review by athanatos — APPROVED** [review]

File: src/crimson/osd/ec_backend.h [hunk 1]
```diff
@@ -33,7 +33,6 @@ class ECBackend : public PGBackend
 		      epoch_t min_epoch, epoch_t max_epoch,
 		      std::vector<pg_log_entry_t>&& log_entries) final;
   CollectionRef coll;
-  crimson::os::FuturizedStore::Shard* store;
```
Comment (liu-chunmei) [inline src/crimson/osd/ec_backend.h:36]: in what situation you need PGBackend::store in crimson? finally we should add it back, right?

Comment (rzarzynski) [reply src/crimson/osd/ec_backend.h:36]: Please note this commit removes the **duplication** of `store` from `ECBackend`, it doesn't touch `PGBackend`. The EC support I'm implementing uses the `store` from `PGBackend`, exactly like `ReplicatedBackend` does.

Comment (liu-chunmei) [reply src/crimson/osd/ec_backend.h:36]: so it is not a sharded store for EC in crimson, right?  can your implementation use seastore also?

**Review by Matan-B — APPROVED** [review]

**Comment by Matan-B** [comment]
jenkins test make check

File: src/crimson/osd/ec_backend.h (hunk 1 above)
Comment (rzarzynski) [reply src/crimson/osd/ec_backend.h:36]: No. I think it's literally the same instance (and the same rules) as with `ReplicatedBackend`.

**Comment by Matan-B** [comment]
https://pulpito.ceph.com/matan-2024-01-18_13:55:25-crimson-rados-wip-matanb-crimson-testing-18.1-crimson-only-distro-crimson-smithi/

**Pull request merged**